test/functional/p2p_invalid_messages.py

```diff
@@ -125,6 +129,27 @@ def test_msgtype(self):
             conn.sync_with_ping(timeout=1)
         self.nodes[0].disconnect_p2ps()
 
+    def test_inv_msg_with_signed_int_types(self):
+        self.log.info("Test inv with signed `type` raises argument out of range error")
+        inv_with_signed_type = msg_inv([CInv(t=-MSG_TX, h=1)])  # inv type must be an unsigned integer
+        try:
+            self.nodes[0].add_p2p_connection(P2PInterface()).build_message(inv_with_signed_type)
+        except struct.error as e:
+            assert_equal("argument out of range", str(e))
+        else:
+            raise AssertionError("No exception raised")
+
+        # Repeat, passing signed integer CInv type format to override default.
+        LE_signed_int = "<i"  # Python format chars for little-endian signed int
+        inv_with_signed_type = msg_inv([CInv(t=-MSG_TX, h=1, f=LE_signed_int)])
```

Comment:
```suggestion
        inv_with_signed_type = msg_inv([CInv(t=0xffffffff, h=1)])
```

Serialized -1 as signed integer is identical to 0xffffffff serialized as unsigned integer, so might as well use the latter

Reply:
That was breaking this method in messages.py::CInv with KeyError: 4294967295
```python
    def __repr__(self):
        return "CInv(type=%s hash=%064x)" \
            % (self.typemap[abs(self.type)], self.hash)
```

Reply:
Even when adding type `0xffffffff: "Error"`?

Reply:
Travis seem happy. I have the following minor cleanups in mind but will wait for feedback before pushing.
```diff
test/functional/p2p_invalid_messages.py
@@ -142,12 +142,12 @@ class InvalidMessagesTest(BitcoinTestFramework):
         LE_signed_int = "<i"  # Python format chars for little-endian signed int
         inv_with_signed_type = msg_inv([CInv(t=-MSG_TX, h=1, f=LE_signed_int)])
-        with self.nodes[0].assert_debug_log(["got inv: 0xffffffff 00000"]):
+        with self.nodes[0].assert_debug_log(["got inv: 0xffffffff "]):
             self.nodes[0].add_p2p_connection(P2PInterface()).send_message(inv_with_signed_type)
 
         # Sanity check with valid unsigned inv type and same format override.
         inv_with_unsigned_type = msg_inv([CInv(t=MSG_WTX, h=1, f=LE_signed_int)])
-        with self.nodes[0].assert_debug_log(["got inv: wtx 0000"]):
+        with self.nodes[0].assert_debug_log(["got inv: wtx "]):

test/functional/test_framework/messages.py
@@ -251,7 +251,7 @@ class CInv:
     def __init__(self, t=0, h=0, f="<I"):
         self.type = t
         self.hash = h
-        self.inv_type_format = f  # default format is unsigned int
+        self.inv_type_format = f  # default is little endian unsigned int
```

Reply:
> Even when adding type `0xffffffff: "Error"`?

Didn't think of it :D thanks, trying.

Reply:
Great--that simplifies things.

Reply:
Done, simplified, good idea. LMK if we should drop the first test.